tests/jerry/es2015/array-prototype-find.js

```diff
@@ -0,0 +1,89 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var array1 = [5, 12, 8, 130, 44];
+
+function bigger_than_10 (element) {
+  return element > 10;
+}
+
+function less_than_0 (element) {
+  return element < 0;
+}
+
+assert (array1.find (bigger_than_10) === 12);
+assert (array1.find (less_than_0) === undefined);
+
+var inventory = [
+    {name: 'apples', quantity: 2},
+    {name: 'bananas', quantity: 0},
+    {name: 'cherries', quantity: 5}
+];
+
+function isCherries (fruit) {
+    return fruit.name === 'cherries';
+}
+
+assert(JSON.stringify (inventory.find (isCherries)) === '{"name":"cherries","quantity":5}');
+assert(JSON.stringify (inventory.find (fruit => fruit.name === 'bananas')) == '{"name":"bananas","quantity":0}');
```

Comment:
It would be good to have the arrow function tests as a separate test cases to mineralize the cross feature dependency.